src/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -142,6 +142,104 @@ public StageManagementResource(string stageServiceUri, HttpSource httpSource)
             return result;
         }
 
+        public async Task<StageDetailedView> Get(string id, Common.ILogger log)

```

Comment:
Resources in NuGet.Core take the Logger for each method. It cannot be passed when the resource is created, and usually you don't want that since every call may have a different caller.